lib/core/managementClient.ts

```diff
@@ -532,6 +556,224 @@ export class ManagementClient extends LinkEntity {
+                messageId: this._safelyGetTypedValue(filtersRawData.value[1]),
+                to: this._safelyGetTypedValue(filtersRawData.value[2]),
+                replyTo: this._safelyGetTypedValue(filtersRawData.value[3]),
+                label: this._safelyGetTypedValue(filtersRawData.value[4]),
+                sessionId: this._safelyGetTypedValue(filtersRawData.value[5]),
+                replyToSessionId: this._safelyGetTypedValue(filtersRawData.value[6]),
+                contentType: this._safelyGetTypedValue(filtersRawData.value[7]),
+                userProperties: this._safelyGetTypedValue(filtersRawData.value[8]),
+              };
+            }
+            break;
+          default:
+            break;
+        }
+
+        if (actionsRawData.descriptor.value === descriptorCodes.sqlRuleActionList
+          && Array.isArray(actionsRawData.value)
+          && actionsRawData.value.length) {
+          rule.action = {
+            expression: this._safelyGetTypedValue(actionsRawData.value[0])
+          };
+        }
+
+        rules.push(rule);
+
+      });
+
+      return rules;
+
+    } catch (err) {
+      const error = translate(err);
+      log.error("An error occurred while sending the get rules request to $management " +
+        "endpoint: %O", error);
+      throw error;
+    }
+  }
+
+  /**
+   * Removes the rule on the Subscription identified by the given rule name.
+   * @param ruleName
+   */
+  async removeRule(ruleName: string): Promise<void> {
+    try {
+      const request: AmqpMessage = {
+        body: {
+          "rule-name": types.wrap_string(ruleName)
+        },
+        reply_to: this.replyTo,
+        application_properties: {
+          operation: Constants.operations.removeRule
+        }
+      };
+      request.application_properties![Constants.trackingId] = generate_uuid();
+
+      log.mgmt("[%s] Remove Rule request body: %O.", this._context.namespace.connectionId,
+        request.body);
+      log.mgmt("[%s] Acquiring lock to get the management req res link.",
+        this._context.namespace.connectionId);
+      await defaultLock.acquire(this.managementLock, () => { return this._init(); });
+
+      await this._mgmtReqResLink!.sendRequest(request);
+
+    } catch (err) {
+      const error = translate(err);
+      log.error("An error occurred while sending the remove rule request to $management " +
+        "endpoint: %O", error);
+      throw error;
+    }
+  }
+
+  /**
+   * Adds a rule on the subscription as defined by the given rule name, filter and action
+   * @param ruleName Name of the rule
+   * @param filter A Boolean, SQL expression or a Correlation filter
+   * @param sqlRuleActionExpression Action to perform if the message satisfies the filtering expression
+   */
+  async addRule(ruleName: string, filter: boolean | string | CorrelationFilter, sqlRuleActionExpression?: string): Promise<void> {
+    if (!ruleName || typeof ruleName !== "string") {
```

Comment:
Can an empty string be a valid ruleName? If so then we should only check for `typeof ruleName !== "string"`?

Let's assume that an empty string is a valid ruleName. Do we want the sdk to allow that behavior? We should sync up with SB team and code accordingly.

Reply:
We get a `MessagingError: The specified name ("") is not valid.` error when the rule name is an empty string. Therefore, empty string is not a valid name.